tests/components/generic/conftest.py

```diff
@@ -89,8 +93,17 @@ def mock_create_stream(hass: HomeAssistant) -> _patch[MagicMock]:
     )
 
 
-@pytest.fixture
-async def user_flow(hass: HomeAssistant) -> ConfigFlowResult:
+@pytest.fixture(name="mock_setup_entry")
+def mock_async_setup_entry_fixture() -> _patch:
+    """Mock setup entry."""
+    return patch(
+        "homeassistant.components.generic.async_setup_entry",
+        return_value=True,
+    )
```

Comment:
A common fixture for this is
```py
@pytest.fixture
def mock_setup_entry() -> Generator[AsyncMock]:
    """Override async_setup_entry."""
    with patch(
        "homeassistant.components.airgradient.async_setup_entry",
        return_value=True,
    ) as mock_setup_entry:
        yield mock_setup_entry
```

This way you only need to load the fixture and its already patch. With the current solution, you would need to always do `with mock_setup_entry` (and I personally try to avoid them if possible)

Reply:
It works well, thanks!

Note: That has reduced the code but resulted in a lot of outdenting in facfe9e, so the diff looks a lot bigger. I suggest reviewing with whitespace ignored:
https://github.com/home-assistant/core/pull/134385/commits/facfe9ebf01edd3b0d7f7ac23d821baeef64f38f?w=1